DEVELOPER-GUIDE.md

```diff
@@ -198,3 +198,48 @@ else
    - If you get an error about access being denied, ensure that all other Notebooks are closed and then restart the kernel again as in step 5
 
 8. Now, use the kernel as you normally would. You should see your local changes being used by the extension.
+
+
+### Setup full suite of tests to run
+
+Some tests require additional setup or will be skipped. `JupyterKernel` tests for e.g. are setup to have the same test run against a Jupyter server, directly against the Jupyter kernel over ZMQ and with a simulation of the messages. Jupyter server and Jupyter kernel tests require the following setup or will be skipped while the tests with simulation can be run without additional steps. 
+
+### Run tests with a local Jupyter Server
+
+1. Install [Jupyter server](https://docs.jupyter.org/en/latest/install.html) or [Anaconda](https://www.anaconda.com/products/distribution)
+2. [Install R kernel](https://docs.anaconda.com/anaconda/user-guide/tasks/using-r-language/) for R tests by calling 
+```
+conda install -c r r-irkernel
```

Comment:
can we clarify if this is to be done by any terminal or the conda one?

Reply:
Updated to clarify